lib/apisession/session.ts

```diff
@@ -0,0 +1,103 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import bunyan = require('bunyan');
+import BAPI = require('../blpapi-wrapper');
+import Subscription = require('../subscription/subscription');
+import Map = require('../util/map');
+import conf = require('../config');
+
+export = Session;
+
+class Session {
+    // PRIVATE VARIABLES
+    private _blpsess: BAPI.Session;
+    private _logger: bunyan.Logger;
+    private _seconds: number; // Last checked activity timestamp
+    private _expired: boolean = false;
+    private _activeSubscriptions: Map<Subscription> = new Map<Subscription>();
+    private _receivedSubscriptions: Map<Subscription> = new Map<Subscription>();
+
+    // PROPERTIES
+    get seconds(): number {
+        return this._seconds;
+    }
+
+    get expired(): boolean {
+        return this._expired;
+    }
+
+    get activeSubscriptions(): Map<Subscription> {
+        return this._activeSubscriptions;
+    }
+
+    get receivedSubscriptions(): Map<Subscription> {
+        return this._receivedSubscriptions;
+    }
+
+    // PUBLIC VARIABLES
+    public inUse: number = 0;
+    public lastPollId: number = undefined;
+    public lastSuccessPollId: number = undefined;
```

Comment:
s/`undefined`/`null` for these two lines.  If any of these are read-only, I would recommend that we make these property getters.